test/box-luatest/gh_8733_ddl_memory_leaks_test.lua

```diff
@@ -0,0 +1,52 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new()
+    cg.server:start()
+    cg.server:exec(function()
+        local s = box.schema.space.create('s')
+        s:create_index('pk', {parts = {{1, 'scalar'}}})
+        s:create_index('sk', {parts = {{1, 'scalar'}}, hint = true})
+        -- We need at least `MEMTX_DDL_YIELD_LOOPS` tuples.
+        for i = 1, 1e3 do
+            s:insert{i}
+        end
+    end)
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+-- Checks that memtx space index build does not leak on multi-statement
+-- transaction error.
```

Comment:
How does this test check that there's no leak?

Reply:
Dropped this test.